### PR TITLE
Move weapon stats to using attributes

### PR DESCRIPTION
**Pull request by Prospector**

### CODE REVIEW

File: src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java [hunk 1]
```diff
@@ -131,62 +144,112 @@ private static void finishWeaponTooltip(ItemStack stack) {
...
+                        EntityAttributes.GENERIC_ATTACK_SPEED,
+                        new EntityAttributeModifier(
+                                SWING_SPEED_MODIFIER_ID, "Attack speed modifier",
+                                ((DgMeleeWeapon) item).getSwingSpeed(stack) - 4.0,
+                                EntityAttributeModifier.Operation.ADDITION
+                        ), slot
+                );
+            }
+        }
+        if (item instanceof DgRangedWeapon) {
```
Comment (Restioson) [inline src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java]: Could this become a method on DgRangedWeapon and similar for the others?

File: src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java [hunk 2]
```diff
@@ -131,62 +144,112 @@ private static void finishWeaponTooltip(ItemStack stack) {
...
+                        } else if (modifier.getId().equals(SWING_SPEED_MODIFIER_ID)) {
+                            value += 4.0;
+                            translationKey = DgEntityAttributes.translationKeyOf("generic.swing_speed");
+                        }
 
-            for (DgWeaponStat stat : stats) {
-                double value = stat.getValue();
-                addLore(stack, new TranslatableText(stat.getTranslationKey(), (value >= 0 ? "+" : "") + format.format(value)).styled(style -> style.withItalic(false).withColor(Formatting.BLUE)));
+                        double g;
+                        if (modifier.getOperation() != EntityAttributeModifier.Operation.MULTIPLY_BASE && modifier.getOperation() != EntityAttributeModifier.Operation.MULTIPLY_TOTAL) {
```
Comment (Restioson) [inline src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java]: What is this doing?

File: src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java [hunk 3]
```diff
@@ -39,8 +50,10 @@
     private static final String DRAW_TIME = "dungeons:draw_time";
 
     public static final Random RANDOM = new Random();
-    private static final UUID ATTACK_SPEED_MODIFIER_ID = UUID.fromString("da3f4761-b0ef-4fdb-a55f-b4269c804055");
-    private static final UUID ATTACK_DAMAGE_MODIFIER_ID = UUID.fromString("ad949c2b-e696-4e50-b98e-325978db25f9");
+    private static final UUID MELEE_DAMAGE_MODIFIER_ID = UUID.fromString("CB3F55D3-645C-4F38-A497-9C13A33DB5CF");
+    private static final UUID SWING_SPEED_MODIFIER_ID = UUID.fromString("FA233E1C-4180-4865-B01B-BCCE9785ACA3");
```
Comment (Restioson) [inline src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java]: nit: this is uppercase and the others are lowercase

File: src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java (hunk 2 above)
Comment (Prospector) [reply src/main/java/xyz/nucleoid/dungeons/dungeons/util/item/DgItemUtil.java]: This is just a modified reimplementation of vanilla's attribute tooltip code

**Comment by Restioson** [comment]
Another thing to consider is making the UUIDs different for when it stacks (what @Patbox said in discord)

**Comment by Prospector** [comment]
Made some changes

> Another thing to consider is making the UUIDs different for when it stacks (what Patbox said in discord)

Yeah so when we add armor, we'll use a different UUID for adjusting melee damage.

File: src/main/java/xyz/nucleoid/dungeons/dungeons/entity/attribute/DgEntityAttributes.java [hunk 4]
```diff
@@ -20,7 +20,7 @@ public static void register() {
 
     private static <T extends EntityAttribute> T register(String id, T attribute) {
         Registry.register(Registry.ATTRIBUTE, new Identifier(Dungeons.ID, id), attribute);
-        return attribute;
+        return attribute;
```
Comment (Restioson) [inline src/main/java/xyz/nucleoid/dungeons/dungeons/entity/attribute/DgEntityAttributes.java]: nit: whitespace

File: src/main/resources/dungeons.mixins.json [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+  "required": true,
+  "minVersion": "0.8",
+  "package": "xyz.nucleoid.dungeons.dungeons.mixin",
+  "compatibilityLevel": "JAVA_8",
+  "mixins": [
+  ],
+  "injectors": {
+    "defaultRequire": 1
+  }
+}
```
Comment (Restioson) [inline src/main/resources/dungeons.mixins.json]: needs newline

File: src/main/resources/assets/dungeons/lang/en_us.json [hunk 6]
```diff
@@ -124,10 +124,10 @@
...
-  "stat.dungeons.swing_speed": "%s Swing Speed",
-  "stat.dungeons.ranged_damage": "%s Ranged Damage",
-  "stat.dungeons.draw_time": "%ss Draw Time",
+  "attribute.dungeons.generic.melee_damage": "Melee Damage",
+  "attribute.dungeons.generic.swing_speed": "Swing Speed",
+  "attribute.dungeons.generic.ranged_damage": "Ranged Damage",
+  "attribute.dungeons.generic.draw_time": "Draw Time",
 
   "": ""
 }
```
Comment (Restioson) [inline src/main/resources/assets/dungeons/lang/en_us.json]: ditto

**Review by Restioson — DISMISSED** [review]
other than the 2 nitpicks lgtm!

**Review by Restioson — APPROVED** [review]

**Pull request merged**